CMakeLists.txt

```diff
@@ -180,7 +181,7 @@ ExternalProject_Add(absl
   EXCLUDE_FROM_ALL ON
   DOWNLOAD_COMMAND ""
   CMAKE_CACHE_ARGS
-    -DCMAKE_CXX_STANDARD:STRING=${THIRD_PARTY_CMAKE_CXX_STANDARD}
+    -DCMAKE_CXX_STANDARD:STRING=${TRITON_MIN_CXX_STANDARD}
```

Comment:
I think we should be propagating `CMAKE_CXX_STANDARD` rather than `TRITON_MIN_CXX_STANDARD`.

Reply:
Could you please elaborate? Since for this set of PRs, I introduce `TRITON_MIN_CXX_STANDARD`, so I use it to set `CMAKE_CXX_STANDARD` for other third_party packages (note: not all of them take it).

Reply:
My understanding was that `TRITON_MIN_CXX_STANDARD` determines the minimum required standard when compiling the project (i.e., since we'll start using C++ X version features). CMAKE_CXX_STANDARD determines the version that the server is actually compiled with. Based on this understanding, I think we should propagate CMAKE_CXX_STANDARD and not `TRITON_MIN_CXX_STANDARD`. Feel free to correct me if I'm wrong.

Reply:
You are correct. However, there are multiple ways how to specify c++ standard for compilation. In `third_party` library I set each library's `CMAKE_CXX_STANDARD` to be equal to `TRITON_MIN_CXX_STANDARD`, since `TRITON_MIN_CXX_STANDARD` is used throughout our triton libraries.

I am not sure if it makes sense to explicitly set CMAKE_CXX_STANDARD here, since we do not compile any custom code, and third parties will use provided ` CMAKE_CXX_STANDARD` through `-DCMAKE_CXX_STANDARD:STRING=${TRITON_MIN_CXX_STANDARD}`.

In other repositories, `target_compile_features(... PRIVATE cxx_std_${TRITON_MIN_CXX_STANDARD})` will make sure that libraries are compiled at least with the provided standard, and if newer standard is needed, then compiler will add appropriate flag, eg `-std=gnu++20`. 

Since I don't introduce hard set of `CMAKE_CXX_STANDARD` throughout  Triton's `server`, `core`, `backend`, `common` libraries, I didn't see the reason why we would need to set it here. And my logic was, since Triton requires at least `TRITON_MIN_CXX_STANDARD` standard, third parties should be compiled with the same standard as minimum required. Note, that some third-parties do not take `CMAKE_CXX_STANDARD`.